File: leetcode/can-make-arithmetic-progression-from-sequence.py
from typing import List


class Solution:
    def canMakeArithmeticProgression(self, arr: List[int]) -> bool:
        length = len(arr)

        arr.sort()
        diff = arr[1] - arr[0]

        for i in range(2, length):
            if arr[i] - arr[i - 1] != diff:
                return False

        return True

File: leetcode/can-make-arithmetic-progression-from-sequence.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
 public:
    bool canMakeArithmeticProgression(vector<int>& arr)
    {
        int diff;

        sort(arr.begin(), arr.end());

        diff = arr[1] - arr[0];

        for (auto i = 2; i < arr.size(); ++i) {
            if (arr[i] - arr[i - 1] != diff) {
                return false;
            }
        }

        return true;
    }
};
